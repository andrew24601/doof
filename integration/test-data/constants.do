// Test constants and null handling
// Tests: LOADK, LOADK_NULL, LOADK_INT16, MOVE

function main(): int {
    let num: int = 42;         // LOADK_INT16
    let str: string = "test";  // LOADK
    let nullVal: int | null = null;  // LOADK_NULL
    
    let copy: int = num;       // MOVE
    
    println(num);    // Should print 42
    println(str);    // Should print "test"
    println(copy);   // Should print 42
    
    // Test null comparison
    if (nullVal == null) {     // EQ_OBJECT or special null comparison
        println("is null");
    } else {
        println("not null");
    }
    
    return 0;
}
