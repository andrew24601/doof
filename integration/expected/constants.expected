42
test
42
is null
